wgpu/tests/common/mod.rs

```diff
@@ -340,12 +340,18 @@ fn initialize_adapter() -> (Adapter, SurfaceGuard) {
     let surface_guard;
     let compatible_surface;
 
-    #[cfg(not(all(target_arch = "wasm32", feature = "webgl")))]
+    #[cfg(not(all(
+        target_arch = "wasm32",
+        any(target_os = "emscripten", feature = "webgl")
+    )))]
     {
         surface_guard = SurfaceGuard {};
         compatible_surface = None;
     }
-    #[cfg(all(target_arch = "wasm32", feature = "webgl"))]
+    #[cfg(all(
+        target_arch = "wasm32",
+        any(target_os = "emscripten", feature = "webgl")
+    ))]
```

Comment:
`create_surface_from_canvas` is not available on `emscripten`, not introduced by this PR but would be nice to solve.

Reply:
I am assuming that the tests aren't run for `emscripten`, should I remove all the `cfg`s checks for it in these tests then?

Reply:
Not currently, but they definitely could be in the future. If it's an arduous maintenance burden, just pretend like they aren't.

Reply:
I'm just gonna leave it as it is now, if that's alright.